### PR TITLE
Add session API calls

### PR DESCRIPTION
**Pull request by jreij**

### CODE REVIEW

File: sessions-core/src/main/java/com/adyen/checkout/sessions/repository/SessionRepository.kt [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            val request = SessionSetupRequest(session.sessionData.orEmpty(), order)
+            return SessionSetupConnection(
+                request = request,
+                environment = configuration.environment,
+                sessionId = session.id,
+                clientKey = configuration.clientKey
+            ).suspendedCall()
+        } catch (e: IOException) {
+            Logger.e(TAG, "SessionSetupConnection Failed", e)
+            throw CheckoutException("Unable to setup session")
```
Comment (caiofaustino) [inline sessions-core/src/main/java/com/adyen/checkout/sessions/repository/SessionRepository.kt:64]: So I've been thinking we should start having some better error handling in general, this looks like a good place to start, can we have a chat about it?

**Comment by caiofaustino** [comment]
I saw you separated the commits based on the type of call (balance, order, payments, details), do you think we should also do that with the model packages? Instead of `request` and `response` we can have them by feature, and have both the request and the response in the same package. What do you think?
`components-core` already kind of does that too

File: sessions-core/src/main/java/com/adyen/checkout/sessions/repository/SessionRepository.kt (hunk 1 above)
Comment (jreij) [reply sessions-core/src/main/java/com/adyen/checkout/sessions/repository/SessionRepository.kt:64]: Yes indeed, let's discuss it and make another PR to refactor all connections

**Comment by jreij** [comment]
> I saw you separated the commits based on the type of call (balance, order, payments, details), do you think we should also do that with the model packages? Instead of `request` and `response` we can have them by feature, and have both the request and the response in the same package. What do you think? `components-core` already kind of does that too

Yeah sure, this is only the example app anyway but I agree as long as we're consistent 👍

**Review by caiofaustino — APPROVED** [review]

**Pull request merged**